src/test/java/org/folio/rest/api/OaiPmhViewTest.java

```diff
@@ -385,6 +402,74 @@ public void testResponseStatus400WhenRequestingWithInvalidUntilDate()
     });
   }
 
+  @Test
+  public void test500(TestContext testContext) {
+    RoutingContext routingContext = mock(RoutingContext.class);
+    when(routingContext.response()).thenReturn(mock(HttpServerResponse.class));
+    AbstractInstanceRecordsAPIImpl.fetchRecordsByQuery("SELECT 1",
+        routingContext, null, testContext.asyncAssertSuccess(response -> {
+          assertThat(response.getStatus(), is(500));
+        }));
+  }
+
+  @Test
+  public void testTcpClose(TestContext testContext) {
```

Comment:
Under what circumstances should the response be closed?

From the investigation I've done, this is triggered by a null pointer exception thrown because of the null parameter. However the setup includes setting that the response headers had already been written.

If a non-null value is used for the parameter (using an empty tuple like in the other tests) the expected behaviour does not occur.

Which scenario was this test intending to describe? Is it that the response is closed when an error occurs and the response has already been (partially) written?